From 562c1b7965daf5588d75a7c5c1a9591a6f871951 Mon Sep 17 00:00:00 2001
From: Chanh Nguyen <chanh@os.amperecomputing.com>
Date: Mon, 26 Apr 2021 14:07:15 +0700
Subject: [PATCH] aspeed: support init GPIOAC2, GPIOAC3, GPIOB0

Initialize the GPIO for Boot EEPROM and Host SPI-NOR mux switch
in A/C power cycle only, no change in BMC reset as these GPIOs 
should be handled by the appropriate features.

Signed-off-by: Chanh Nguyen <chanh@os.amperecomputing.com>
---
 board/aspeed/ast-g5/ast-g5.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/board/aspeed/ast-g5/ast-g5.c b/board/aspeed/ast-g5/ast-g5.c
index edf7f050a8..f1ada1527b 100644
--- a/board/aspeed/ast-g5/ast-g5.c
+++ b/board/aspeed/ast-g5/ast-g5.c
@@ -119,6 +119,27 @@ int board_late_init(void)
 		update_bootargs_cmd("resetreason", "watchdog");
 	} else if (reset_reason & SCU_SYS_PWR_RESET_FLAG) {
 		update_bootargs_cmd("resetreason", "power");
+
+		/* Init GPIO after cold bmc boot */
+
+		/*
+		 * BMC_GPIOAC2_SPI0_PROGRAM_SEL
+		 * HOST control SPI-Host and EEPROM
+		 */
+		gpio_direction_output(226, 0);
+
+		/*
+		 * BMC_SPI0_BACKUP_SEL - GPIOAC3
+		 * Boot from MAIN SPI-HOST
+		 */
+		gpio_direction_output(227, 0);
+
+		/*
+		 * BMC_GPIOB0_I2C_BACKUP_SEL
+		 * Boot from MAIN EEPROM
+		 */
+		gpio_direction_output(8, 1);
+
 	} else {
 		char value[32];
 		snprintf(value, sizeof(value) - 1, "0x%x", reset_reason);
-- 
2.17.1

